src/libostree/ostree-bootloader-grub2.c

```diff
@@ -299,8 +299,25 @@ _ostree_bootloader_grub2_write_config (OstreeBootloader      *bootloader,
   g_autofree char *bootversion_str = g_strdup_printf ("%u", (guint)bootversion);
   g_autoptr(GFile) config_path_efi_dir = NULL;
   g_autofree char *grub2_mkconfig_chroot = NULL;
+  gboolean use_grub2_mkconfig = FALSE;
+  const gchar *grub_exec = NULL;
+
+#ifdef OSTREE_GRUB2_MKCONFIG
+  use_grub2_mkconfig = TRUE;
+#endif
+  /* Autotests can set this envvar to select which code path to test, useful for OS installers as well */
+  grub_exec = g_getenv ("OSTREE_GRUB2_EXEC");
+  if (grub_exec)
+    {
+      if (g_str_has_suffix(grub_exec, "-mkconfig"))
+        use_grub2_mkconfig = TRUE;
```

Comment:
The reason I decided to check only for the "-mkconfig" suffix is because on Ubuntu there is no "grub2-mkconfig". On Ubuntu it is simply called grub-mkconfig, so "-mkconfig" would work for both distros then. I was thinking about adding a comment in the source code about this.